File: Examples/2_example/Java/controllers/UserController.java
package controllers;

import models.UserModel;
import views.UserView;
import java.util.List;

public class UserController {
  private UserModel userModel;
  private UserView userView;

  public UserController(){
    this.userModel = new UserModel();
    this.userView = new UserView();
  }

  public String list() {
    List<String> users = userModel.getAll();
    return userView.listUsers(users);
  }

  public String add(String user) {
    userModel.add(user);
    return userView.addUser(user);
  }
}

File: Examples/Example_2/TypeScript/controllers/UserController.ts
import { Request, Response } from "express";
import { UserModel } from "../models/UserModel";
import { UserView } from "../views/UserView";

export class UserController {
  private userModel: UserModel = new UserModel();
  private userView: UserView = new UserView();

  list(req:Request, res:Response): void {
    const users = this.userModel.getAll();

    res.send(this.userView.listUsers(users));
  }

  add(req:Request, res:Response): void {
    const user = req.query.name as string;
    this.userModel.add(user);

    res.send(this.userView.addUser(user));
  }
}
